catalog/core/catalog-core-api/src/main/java/ddf/catalog/plugin/AccessPlugin.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.plugin;
+
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.UpdateRequest;
+
+/**
+ * An AccessPlugin allows or denies access to the Catalog operation or response.
```

Comment:
To the first question: that is one way, but not necessarily the only way. The plugin could also remove records from a response, for instance. Or modify those records in some way so that the receiver could not get info from them. To the second question: I'm not really sure I know what you mean by calling them in reverse order?